app/services/school_data_importer.rb

```diff
@@ -0,0 +1,70 @@
+require "open-uri"
+require "csv"
+
+class SchoolDataImporter
+  def run
+    download_and_save_csv
+
+    CSV.foreach(csv_file_name, headers: true, encoding: "windows-1251:utf-8") do |row|
+      local_authority = LocalAuthority.find_or_create_by(
+        code: row["LA (code)"],
+        name: row["LA (name)"]
+      )
+
+      school = row_to_school(row)
+
+      school.local_authority = local_authority
+
+      unless school.valid?
+        Rails.logger.warn("Failed to import school: #{school.urn} - #{school.errors.full_messages}") && next
+      end
+      school.save
+    end
+
+    delete_csv_file
+  rescue OpenURI::HTTPError => e
+    Rails.logger.error("Get Information About Schools request: Unsuccessful #{e.message}")
+  rescue Net::OpenTimeout
+    Rails.logger.error("Get Information About Schools request: Timeout")
+  end
+
+  private
+
+  def row_to_school(row)
+    school = School.find_or_initialize_by(urn: row["URN"])
+    school.name = row["EstablishmentName"]
+    school.street = row["Street"]
+    school.locality = row["Locality"]
+    school.town = row["Town"]
+    school.county = row["County (name)"]
+    school.postcode = row["Postcode"]
+    school.phase = row["PhaseOfEducation (code)"].to_i
+    school.school_type_group_code = row["EstablishmentTypeGroup (code)"]
+    school.school_type_code = row["TypeOfEstablishment (code)"]
+    school
+  end
+
+  def download_and_save_csv
+    File.open(csv_file_name, "wb") do |saved_file|
```

Comment:
Rather than writing to disk explicitly we could simply return the file handle that `open` returns. That way we don't have to worry about cleaning up tempfiles ourselves as Ruby will do it for us.